src/couch_replicator_httpc_pool.erl

```diff
@@ -31,8 +31,7 @@
 -record(state, {
     url,
     limit,                  % max # of workers allowed
-    free = [],              % free workers (connections)
-    busy = [],              % busy workers (connections)
+    conns = [],
```

Comment:
Yeah, my line of thought was that httpc_connection_pool could be setting a connection cap whereas the connection sharing bit would be unbounded. The logic to set a cap on the total number of connections would cause the logic of handling that in the per-replication pooling a fair bit more complicated.